diff --git a/es/config-provider/cssVariables.js b/es/config-provider/cssVariables.js
index 896d15ae169493475835fe1771e7fe362d142d2f..9bb0a340db5418567cd562a4da446b7e544265ac 100644
--- a/es/config-provider/cssVariables.js
+++ b/es/config-provider/cssVariables.js
@@ -77,7 +77,7 @@ export function registerTheme(globalPrefixCls, theme) {
     return "--".concat(globalPrefixCls, "-").concat(key, ": ").concat(variables[key], ";");
   });
   if (canUseDom()) {
-    updateCSS("\n  :root {\n    ".concat(cssList.join('\n'), "\n  }\n  "), "".concat(dynamicStyleMark, "-dynamic-theme"));
+    updateCSS("\n  html {\n    ".concat(cssList.join('\n'), "\n  }\n  "), "".concat(dynamicStyleMark, "-dynamic-theme"));
   } else {
     devWarning(false, 'ConfigProvider', 'SSR do not support dynamic theme with css variables.');
   }
diff --git a/es/input/Input.js b/es/input/Input.js
index b82d75140e62213973551b10a852a2ce194f3e83..4bd6d95debd9c349250c092e40e5b85164acb8e8 100644
--- a/es/input/Input.js
+++ b/es/input/Input.js
@@ -192,6 +192,7 @@ export default defineComponent({
       setValue('', function () {
         focus();
       });
+      emit('clear');
     };
     var handleChange = function handleChange(e) {
       var _e$target = e.target,
diff --git a/es/select/index.d.ts b/es/select/index.d.ts
index ab32f2f69095db186025fffb39b6058b46766c2f..32105b6cc5eda6c91ab06a37d01066ef76161680 100644
--- a/es/select/index.d.ts
+++ b/es/select/index.d.ts
@@ -106,7 +106,7 @@ export declare const selectProps: () => {
     onInputKeyDown: PropType<(e: KeyboardEvent) => void>;
     removeIcon: import("vue-types").VueTypeValidableDef<any>;
     maxTagCount: {
-        type: PropType<number | "responsive">;
+        type: PropType<number | "responsive"| 'gct-responsive'>;
     };
     maxTagTextLength: NumberConstructor;
     maxTagPlaceholder: import("vue-types").VueTypeValidableDef<any>;
diff --git a/es/table/Table.js b/es/table/Table.js
index 0a9219f79f8c66eda23b948fcf7aa56ff28045ad..c1b8efd530ad12b9a924346c14c7dd8407989c6c 100644
--- a/es/table/Table.js
+++ b/es/table/Table.js
@@ -233,6 +233,7 @@ var InteralTable = defineComponent({
     }));
     useProvideTableContext({
       onResizeColumn: function onResizeColumn(w, col) {
+        col.width=w
         emit('resizeColumn', w, col);
       }
     });
diff --git a/es/tree-select/index.d.ts b/es/tree-select/index.d.ts
index fa7785f37df464bbaec380765da10e84523fa22e..769f90f8036a83db9cf79f128381c197156a2174 100644
--- a/es/tree-select/index.d.ts
+++ b/es/tree-select/index.d.ts
@@ -384,7 +384,7 @@ declare const _default: {
             onInputKeyDown: PropType<(e: KeyboardEvent) => void>;
             removeIcon: import("vue-types").VueTypeValidableDef<any>;
             maxTagCount: {
-                type: PropType<number | "responsive">;
+                type: PropType<number | "responsive"| 'gct-responsive'>;
             };
             maxTagTextLength: NumberConstructor;
             maxTagPlaceholder: {
diff --git a/es/vc-overflow/Overflow.js b/es/vc-overflow/Overflow.js
index f58a9e52412c9c1682d2b55eee8c926b1c95d7ed..1ce6a1cd43a78b019d4ab7d1c6a399a60200c136 100644
--- a/es/vc-overflow/Overflow.js
+++ b/es/vc-overflow/Overflow.js
@@ -10,6 +10,7 @@ import { OverflowContextProvider } from './context';
 import Item from './Item';
 import RawItem from './RawItem';
 var RESPONSIVE = 'responsive';
+var GCT_RESPONSIVE = 'gct-responsive';
 var INVALIDATE = 'invalidate';
 function defaultRenderRest(omittedItems) {
   return "+ ".concat(omittedItems.length, " ...");
@@ -80,7 +81,7 @@ var Overflow = defineComponent({
     });
     // ================================= Data =================================
     var isResponsive = computed(function () {
-      return !!(props.data.length && props.maxCount === RESPONSIVE);
+      return !!(props.data.length && (props.maxCount === RESPONSIVE||props.maxCount ===GCT_RESPONSIVE));
     });
     var invalidate = computed(function () {
       return props.maxCount === INVALIDATE;
@@ -170,7 +171,7 @@ var Overflow = defineComponent({
         for (var i = 0; i < len; i += 1) {
           var currentItemWidth = getItemWidth(i);
           // Break since data not ready
-          if (currentItemWidth === undefined) {
+          if (currentItemWidth === undefined&&(props.maxCount!==GCT_RESPONSIVE||i>4)) {
             updateDisplayCount(i - 1, true);
             break;
           }
@@ -185,7 +186,7 @@ var Overflow = defineComponent({
             updateDisplayCount(lastIndex);
             suffixFixedStart.value = null;
             break;
-          } else if (totalWidth + mergedRestWidth.value > mergedContainerWidth.value) {
+          } else if ((totalWidth + mergedRestWidth.value > mergedContainerWidth.value)&&(props.maxCount!==GCT_RESPONSIVE||i>4)) {
             // Can not hold all the content to show rest
             updateDisplayCount(i - 1);
             suffixFixedStart.value = totalWidth - currentItemWidth - suffixWidth.value + restWidth.value;
@@ -323,4 +324,5 @@ var Overflow = defineComponent({
 Overflow.Item = RawItem;
 Overflow.RESPONSIVE = RESPONSIVE;
 Overflow.INVALIDATE = INVALIDATE;
+Overflow.GCT_RESPONSIVE=GCT_RESPONSIVE
 export default Overflow;
\ No newline at end of file
diff --git a/es/vc-select/Selector/MultipleSelector.js b/es/vc-select/Selector/MultipleSelector.js
index 8696d776c116dc7047dce10c9609634400e06ef9..1e6fda4ddb519049b3e5d14a1dc37fc56a8502d8 100644
--- a/es/vc-select/Selector/MultipleSelector.js
+++ b/es/vc-select/Selector/MultipleSelector.js
@@ -109,7 +109,8 @@ var SelectSelector = defineComponent({
     }
     function customizeRenderSelector(value, content, itemDisabled, closable, onClose, option) {
       var onMouseDown = function onMouseDown(e) {
-        onPreventMouseDown(e);
+        // onPreventMouseDown(e);
+        e.stopPropagation();
         props.onToggleOpen(!open);
       };
       var originData = option;
diff --git a/es/vc-select/Selector/index.js b/es/vc-select/Selector/index.js
index bb0bb3ad071230d515db41cd676d6da0395c557f..8e93e08396fa4f74065247b1120ffd34fdf5d668 100644
--- a/es/vc-select/Selector/index.js
+++ b/es/vc-select/Selector/index.js
@@ -166,7 +166,7 @@ var Selector = defineComponent({
     var onMousedown = function onMousedown(event) {
       var inputMouseDown = getInputMouseDown();
       if (event.target !== inputRef.current && !inputMouseDown) {
-        event.preventDefault();
+        // event.preventDefault();
       }
       if (props.mode !== 'combobox' && (!props.showSearch || !inputMouseDown) || !props.open) {
         if (props.open) {
diff --git a/es/vc-table/Header/HeaderRow.js b/es/vc-table/Header/HeaderRow.js
index d62239e1a17df39949f1737059b4a8619f035b84..914929d264d68a0ffc663c4792e545d1ea05a03f 100644
--- a/es/vc-table/Header/HeaderRow.js
+++ b/es/vc-table/Header/HeaderRow.js
@@ -56,7 +56,7 @@ export default defineComponent({
                 return column.title;
               },
               dragHandle: function dragHandle() {
-                return col.resizable ? _createVNode(DragHandleVue, {
+                return col.resizable!==false ? _createVNode(DragHandleVue, {
                   "prefixCls": prefixCls,
                   "width": col.width,
                   "minWidth": col.minWidth,
