diff --git a/.vscode/settings.json b/.vscode/settings.json
new file mode 100644
index 0000000000000000000000000000000000000000..243453ed55350e88b341bec7b29ecdc8866e73df
--- /dev/null
+++ b/.vscode/settings.json
@@ -0,0 +1,11 @@
+{
+    "i18n-ally.localesPaths": [
+        "es/locale",
+        "lib/locale",
+        "packages/locale",
+        "es/locale/lang",
+        "lib/locale/lang",
+        "packages/locale/lang"
+    ],
+    "vue-i18n.i18nPaths": "es\\locale,lib\\locale,packages\\locale,es\\locale\\lang,lib\\locale\\lang,packages\\locale\\lang"
+}
\ No newline at end of file
diff --git a/es/edit/src/hook.js b/es/edit/src/hook.js
index e982fdb9f387bd4d16949bc1090c5a04b822fbea..7d264372bc66960cf1855113653cd080e716db73 100644
--- a/es/edit/src/hook.js
+++ b/es/edit/src/hook.js
@@ -518,32 +518,65 @@ const editHook = {
              * 清除激活的编辑
              */
             clearEdit(evnt) {
-                const { editStore } = reactData;
-                const { actived, focused } = editStore;
-                const { row, column } = actived;
+              const editOpts = computeEditOpts.value
+              const { editStore } = reactData
+              const { actived, focused } = editStore
+              const { row, column } = actived
+              const { beforeClearEdit } = editOpts
+
+              const clearEditAction = () => {
                 if (row || column) {
-                    syncActivedCell();
-                    actived.args = null;
-                    actived.row = null;
-                    actived.column = null;
-                    $xetable.updateFooter();
-                    $xetable.dispatchEvent('edit-closed', {
-                        row,
-                        rowIndex: $xetable.getRowIndex(row),
-                        $rowIndex: $xetable.getVMRowIndex(row),
-                        column,
-                        columnIndex: $xetable.getColumnIndex(column),
-                        $columnIndex: $xetable.getVMColumnIndex(column)
-                    }, evnt || null);
+                  syncActivedCell()
+                  actived.args = null
+                  actived.row = null
+                  actived.column = null
+                  $xetable.updateFooter()
+                  $xetable.dispatchEvent('edit-closed', {
+                    row,
+                    rowIndex: $xetable.getRowIndex(row),
+                    $rowIndex: $xetable.getVMRowIndex(row),
+                    column,
+                    columnIndex: $xetable.getColumnIndex(column),
+                    $columnIndex: $xetable.getVMColumnIndex(column)
+                  }, evnt || null)
                 }
                 if (GlobalConfig.cellVaildMode === 'obsolete') {
-                    if ($xetable.clearValidate) {
-                        return $xetable.clearValidate();
+                  if ($xetable.clearValidate) {
+                    return $xetable.clearValidate()
+                  }
+                }
+                focused.row = null
+                focused.column = null
+                return nextTick()
+              }
+
+              if (row || column) {
+                if (beforeClearEdit) {
+                  const result = beforeClearEdit({
+                    row,
+                    rowIndex: $xetable.getRowIndex(row),
+                    column,
+                    columnIndex: $xetable.getColumnIndex(column),
+                    $table: $xetable,
+                    $grid: $xetable.xegrid
+                  })
+
+                  if (result instanceof Promise) {
+                    return result.then((shouldClear) => {
+                      if (shouldClear !== false) {
+                        return clearEditAction()
+                      }
+                      return nextTick()
+                    })
+                  } else {
+                    if (result === false) {
+                      return nextTick()
                     }
+                  }
                 }
-                focused.row = null;
-                focused.column = null;
-                return nextTick();
+              }
+
+              return clearEditAction()
             },
             /**
              * 清除所选中源状态
diff --git a/es/table/src/header.js b/es/table/src/header.js
index 04aaf9bdd69792ff9f61f329d7596920bea958ed..445174fd42c7f86a9f255ec010e8f8fbd1b86c71 100644
--- a/es/table/src/header.js
+++ b/es/table/src/header.js
@@ -67,7 +67,8 @@ export default defineComponent({
                     tempCellElem = tempCellElem[siblingProp];
                 }
                 if (isRightFixed && rightContainerElem) {
-                    dragPosLeft = rightContainerElem.offsetLeft + fixedOffsetWidth;
+                    dragPosLeft = rightContainerElem.offsetLeft + fixedOffsetWidth+cell.offsetWidth
+                    dragMinLeft = rightContainerElem.offsetLeft + fixedOffsetWidth + minInterval
                 }
             }
             // 处理拖动事件
@@ -83,8 +84,8 @@ export default defineComponent({
                 }
                 else if (isRightFixed) {
                     // 右侧固定列（不允许超过左侧固定列、不允许超过左边距）
-                    dragMinLeft = (leftContainerElem ? leftContainerElem.clientWidth : 0) + fixedOffsetWidth + minInterval;
-                    left = Math.min(left, dragPosLeft + cell.clientWidth - minInterval);
+                    // dragMinLeft = (leftContainerElem ? leftContainerElem.clientWidth : 0) + fixedOffsetWidth + minInterval;
+                    // left = Math.min(left, dragPosLeft + cell.clientWidth - minInterval);
                 }
                 else {
                     dragMinLeft = Math.max(tableBodyElem.scrollLeft, dragMinLeft);
@@ -100,7 +101,9 @@ export default defineComponent({
             document.onmouseup = function (evnt) {
                 document.onmousemove = domMousemove;
                 document.onmouseup = domMouseup;
-                const resizeWidth = column.renderWidth + (isRightFixed ? dragPosLeft - dragLeft : dragLeft - dragPosLeft);
+                // const resizeWidth = column.renderWidth + (isRightFixed ? dragPosLeft - dragLeft : dragLeft - dragPosLeft);
+                //新版逻辑忽略右冻结处理
+                const resizeWidth = column.renderWidth + dragLeft - dragPosLeft
                 column.resizeWidth = resizeWidth;
                 resizeBarElem.style.display = 'none';
                 tableReactData._isResize = false;
diff --git a/es/table/src/table.js b/es/table/src/table.js
index 61aa6c067ed5b2f1dd6626c2bfba65b9924eaaa4..1ecea70b2ea921fffca7a4b74e67c1947b585e60 100644
--- a/es/table/src/table.js
+++ b/es/table/src/table.js
@@ -6733,6 +6733,16 @@ export default defineComponent({
                         resizeObserver.observe(parentEl);
                     }
                 }
+                const headerEl=refTableHeader.value.$el
+                const borderEl=refTableBody.value.$el
+                headerEl.onwheel=(e)=>{
+                    if(!reactData.overflowX) return 
+                    const {currentTarget,deltaX}=e
+                    const scrollLeft = currentTarget.scrollLeft + deltaX
+                    currentTarget.scrollLeft=scrollLeft
+                    borderEl.scrollLeft=scrollLeft
+                    e.preventDefault();
+                }
             });
             GlobalEvent.on($xetable, 'paste', handleGlobalPasteEvent);
             GlobalEvent.on($xetable, 'copy', handleGlobalCopyEvent);
diff --git a/types/table.d.ts b/types/table.d.ts
index c659ba467026cd8b464cff3c37a2ed5dd07c41f1..ff847b7119fbe4245585d2ef36cb7209cb503403 100644
--- a/types/table.d.ts
+++ b/types/table.d.ts
@@ -2336,7 +2336,17 @@ export namespace VxeTablePropTypes {
       $table: VxeTableConstructor<DT> & VxeTablePrivateMethods<DT>
       $grid: VxeGridConstructor<DT> | null | undefined
     }): void
-
+    /**
+     * 清除编辑之前的回调，该方法返回 false 则阻止清除编辑状态
+     */
+    beforeClearEdit?(params: {
+      row: DT
+      rowIndex: number
+      column: VxeTableDefines.ColumnInfo<DT>
+      columnIndex: number
+      $table: VxeTableConstructor<DT> & VxeTablePrivateMethods<DT>
+      $grid: VxeGridConstructor<DT> | null | undefined
+    }): Promise<boolean> | boolean
     /**
      * 请使用 beforeEditMethod
      * @deprecated
